src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -108,6 +130,24 @@ def _schedule_job_to_run_or_block(cls, job: Job):
                 cls.jobs_to_run.put(job)
             cls.__check_and_execute_jobs_if_development_mode()
 
+    @classmethod
+    def __wait_until_job_finished(cls, jobs: Union[List[Job], Job], timeout: Optional[int] = None):
+        def __check_if_timeout(start, timeout):
+            if timeout:
+                return (datetime.now() - start).seconds < timeout
+            return True
+
+        start = datetime.now()
+        jobs = jobs if isinstance(jobs, Iterable) else [jobs]
+
+        while __check_if_timeout(start, timeout) and jobs:
+            try:
+                if jobs[0]._is_finished():
+                    jobs.pop(0)
+                sleep(0.5)  # Limit CPU usage
```

Comment:
```suggestion
                if jobs[0]._is_finished():
                    jobs.pop(0)
                else:
                    sleep(0.5)  # Limit CPU usage
                
```
if the first job to check is finished, we don't need to sleep before trying the next job. (that s basically the third point of my previous comment)

Reply:
ah ha! yep that makes much more sense! :D